### PR TITLE
Cleanup and unify CodeMirror Demo pages

### PR DESCRIPTION
**Pull request by VasylMarchuk**
Related to #1231

### Brief

In preparation for adding `FileDiffCardCM` component (CodeMirror-based clone of `FileDiffCard`) to the Demo Zone, this cleans up & unifies the Demo route pages.

### Details

This only involves the Demo page, nothing else. Changes to `FileContentsCard` Demo page are inspired by the upcoming addition of `FileDiffCardCM` to the Demo, and make all Demo pages behave similarly.

### More details

- Extracted all example Demo documents into `app/utils/code-mirror-documents`, to reuse them across Demo pages instead of cloning:
  - documents in the util are now generated from **diffs**
  - these diffs are much easier to work with than `document` & `originalDocument` pairs
  - example documents can now be used for diff-consuming components (such as `FileDiffCard`)
- Made all options from `FileContentsCard` Demo page persistent via query params
- Added `Reset all options` link to `FileContentsCard` Demo page
- Moved `DarkModeToggle` component to appear last in the Demo Zone navigation bar
- Extracted diff-parsing logic from `diff-to-document` helper into an util `parse-diff-as-document`, to reuse it in `code-mirror-documents` util

### Checklist

- [x] I've thoroughly self-reviewed my changes
- [x] I've added tests for my changes, unless they affect admin-only areas (or are otherwise not worth testing)
- [x] I've verified any visual changes using Percy (add a commit with `[percy]` in the message to trigger)

## Release Notes

- **New Features**
  - Introduced a "Reset all options" button in the file contents card for enhanced interactivity.
  - Added a streamlined management system for example documents using a predefined set instead of dynamic instantiation.

- **Improvements**
  - Enhanced handling of query parameters in the file contents card route for better URL management.
  - Simplified parsing logic in the document helper function for improved performance.

- **Bug Fixes**
  - Adjusted routing and display logic for links to ensure correct navigation between components.

- **Tests**
  - Added new test cases for the `parseDiffAsDocument` function to ensure reliability and error handling.

### CODE REVIEW

File: app/helpers/diff-to-document.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { helper } from '@ember/component/helper';
 
-const diffToDocument = helper(function diffToDocument([diff = '']: [string | undefined]) {
+export function parseDiff(diff: string = '') {
```
Comment (rohitpaulk) [inline app/helpers/diff-to-document.ts:3]: @VasylMarchuk is there a downside to keeping this as it was? I think that's what we do in other helpers. Not a big deal but if we can stay consistent might as well.

File: app/router.ts [hunk 2]
```diff
@@ -104,7 +104,7 @@ Router.map(function () {
 
   this.route('demo', function () {
     this.route('code-mirror');
-    this.route('dark-mode-toggle');
     this.route('file-contents-card');
+    this.route('dark-mode-toggle');
```
Comment (rohitpaulk) [inline app/router.ts:108]: These are usually alphabetically sorted, so the previous version was correct

Comment (rohitpaulk) [reply app/router.ts:108]: (Anywhere order doesn't matter, we try to stick to alphabetical ordering)

**Review by rohitpaulk — APPROVED** [review]

File: app/helpers/diff-to-document.ts (hunk 1 above)
Comment (VasylMarchuk) [reply app/helpers/diff-to-document.ts:3]: I want to reuse the parsing method that the helper uses in a different place (not the helper itself), maybe I can move it into a separate util?

Comment (rohitpaulk) [reply app/helpers/diff-to-document.ts:3]: Ah, gotcha - yep let's move that to lib then! Makes it more obvious that it will be reused

Comment (VasylMarchuk) [reply app/helpers/diff-to-document.ts]: Done, moved into `utils/parse-diff-as-document` + added unit tests (copied from helper integration tests, but in unit syntax)

File: app/utils/parse-diff-as-document.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  for (const line of diffLines) {
+    if (line.startsWith('-')) {
+      original.push(line.substring(1));
+    } else if (line.startsWith('+')) {
+      current.push(line.substring(1));
+    } else {
+      original.push(line.substring(1));
+      current.push(line.substring(1));
+    }
+  }
```
[the remark this answers is not in the document]
Comment (VasylMarchuk) [reply app/utils/parse-diff-as-document.ts:16]: Good idea for a separate PR if this comes up as an issue

File: app/utils/parse-diff-as-document.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  return {
+    current: current.join('\n'),
+    original: original.join('\n'),
+  };
+}
```
[the remark this answers is not in the document]
Comment (VasylMarchuk) [reply app/utils/parse-diff-as-document.ts:22]: Our diffs are not exactly classical diffs, they don't contain headers or hunks

File: app/utils/parse-diff-as-document.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (VasylMarchuk) [reply app/utils/parse-diff-as-document.ts:16]: Considering that our diffs are not really classical diffs, we'll handle this if we ever want to migrate to true diffs.

**Pull request merged**